lib/spack/llnl/util/argparsewriter.py

```diff
@@ -96,7 +105,7 @@ def parse(self, parser, prog):
                     subcommands.append((subparser, subaction.dest))
 
                     # Look for aliases of the form 'name (alias, ...)'
-                    if self.aliases:
+                    if self.aliases and isinstance(subaction.metavar, str):
```

Comment:
Was this caught by `mypy`?

Reply:
Yes! Apparently it can also be None or a list of strings